### ISSUE TITLE
Should have default shape if one is not specified

### ISSUE THREAD

**Issue by emp823** [viewthespace/versioner-rails#7]
If no shape gets passed, there should be a default

**Comment by lifmus** [comment]
+1

**Comment by dnasseri** [comment]
There's a few issues at play here. 

Should we actually be able to specify a default serializer for an endpoint? What exactly is a 'default serializer'. Calling `to_json` on the resource seems like an interesting approach, but might lead to security concerns where we send back fields on an object that, at the application layer, a user might not be able to see. A non-explicit, conventional default serializer such as Foo::FooSerializer could be another approach.

Another concern is that allowing for a default serializer would essentially be a silent failure in many applications. Instead of specifying a default maybe we should raise an error, or at the very least, log a warning.

**Comment by CharlesMcMillan** [comment]
With regards to the 'silent failure' dilemma, perhaps in addition to any warning-level logging, we could find an appropriate status code to send to the client? Does a [300 status code](http://www.checkupdown.com/status/E300.html) fit the bill here?

**Comment by CharlesMcMillan** [comment]
Playing devil's advocate with regards to `to_json` exposing some security risks, I could see that being something the application should handle/protect against. If an application is really concerned about a field being sensitive and requiring additional protections, I'd argue that the application shouldn't include that in `to_json` to be safe (any new engineer could make a mistake by calling it, and there goes all the security measures they've worked so hard to build :smile: )

As a secondary approach, I think this could become configurable. The consumer could specify the preferred default_strategy in a config file...something like:
```ruby
Versioner.configure do |config|
  config.default_strategy = Versioner.TO_JSON_STRATEGY
end
```

OR

```ruby
Versioner.configure do |config|
  config.default_strategy = Versioner.SHAPE_ROOT_FOLDER
end
```

The former would let versioner know to always use to_json and the latter would go about choosing a default serializer (whether specified in the controller or magically found by the gem)

**Comment by shaw3257** [comment]
I agree that we shouldn't default to `to_json`. @CharlesMcMillan I like your configuration approach. It could be used in conjunction with the following proposal:

What if we move away from
``` Ruby
class FoosController < ActionController::Base
  versioner Serializers::Foo

  def show
    render json: Foo.new, serializer: versioner_serializer
  end

end
```
to
``` Ruby
class FoosController < ActionController::Base

  def show
    render json: Foo.new, serializer: versioner_rails(default: Serializers::Foo::AnySerializer)
  end

end
```

We can define `versioner_rails` at load time on all rails controllers and have the option to pass in a default. If a default is provided, and a request contains no shape or version, we use that default. If the client request has a shape or version that can't be found, we throw an error.
